api/v1alpha1/api.go

```diff
@@ -43,22 +46,21 @@ type LLMRouteSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self.schema == 'OpenAI'"
 	APISchema LLMAPISchema `json:"inputSchema"`
-	// TargetRefs are the names of the Gateway resources this policy is being attached to.
+	// HTTPRouteRef is the name of the HTTPRoute resource that the Gateway will use to route the traffic.
```

Comment:
One downside of `HTTPRouteRef` is that we lose the owner reference between LLMRoute and HTTPRoute

Reply:
https://github.com/envoyproxy/ai-gateway/pull/39/commits/eecdbb873b8a20bda944224fb979addb8f120594 agree. as discussed offline, I went ahead and made a change to use HTTPRouteSpec